File: Lab 01/Lab 1 - t1.py
import math

# Определяем комплексные числа
z1 = (1, 1)  # 1 + i
z2 = (3, -1)  # 3 - i

# Сложение
sum_z = (z1[0] + z2[0], z1[1] + z2[1])

# Вычитание
diff_z = (z1[0] - z2[0], z1[1] - z2[1])

# Умножение
prod_z = (z1[0] * z2[0] - z1[1] * z2[1], z1[0] * z2[1] + z1[1] * z2[0])

# Деление
numerator = (z1[0] * z2[0] + z1[1] * z2[1], z1[1] * z2[0] - z1[0] * z2[1])
denominator = z2[0]**2 + z2[1]**2
quot_z = (numerator[0] / denominator, numerator[1] / denominator)

# Четвертая степень и корень третьей степени для -1 + i
z3 = (-1, 1)

# Четвертая степень
r = math.sqrt(z3[0]**2 + z3[1]**2)
theta = math.atan2(z3[1], z3[0])
r4 = r**4
theta4 = 4 * theta
z3_fourth_power = (r4 * math.cos(theta4), r4 * math.sin(theta4))

# Корень третьей степени (для трех корней)
r3 = r**(1/3)
z3_cube_roots = []
for k in range(3):
    theta3 = (theta + 2 * math.pi * k) / 3
    z3_cube_roots.append((r3 * math.cos(theta3), r3 * math.sin(theta3)))

# Выводим результаты
print(f"Сумма: {sum_z[0]} + {sum_z[1]}i")
print(f"Разность: {diff_z[0]} + {diff_z[1]}i")
print(f"Произведение: {prod_z[0]} + {prod_z[1]}i")
print(f"Частное: {quot_z[0]} + {quot_z[1]}i")
print(f"Четвертая степень числа -1+i: {z3_fourth_power[0]} + {z3_fourth_power[1]}i")
print(f"Корни третьей степени из числа -1+i:")
for root in z3_cube_roots:
    print(f"{root[0]} + {root[1]}i")

File: Lab 01/Lab1 - t1.cpp
#include <iostream>
#include <cmath>
#include <vector>

int main() {
    // Определяем комплексные числа
    std::pair<double, double> z1 = {1, 1};  // 1 + i
    std::pair<double, double> z2 = {3, -1};  // 3 - i

    // Сложение
    std::pair<double, double> sum_z = {z1.first + z2.first, z1.second + z2.second};

    // Вычитание
    std::pair<double, double> diff_z = {z1.first - z2.first, z1.second - z2.second};

    // Умножение
    std::pair<double, double> prod_z = {z1.first * z2.first - z1.second * z2.second, z1.first * z2.second + z1.second * z2.first};

    // Деление
    std::pair<double, double> numerator = {z1.first * z2.first + z1.second * z2.second, z1.second * z2.first - z1.first * z2.second};
    double denominator = pow(z2.first, 2) + pow(z2.second, 2);
    std::pair<double, double> quot_z = {numerator.first / denominator, numerator.second / denominator};

    // Четвертая степень и корень третей степени для -1 + i
    std::pair<double, double> z3 = {-1, 1};

    // Четвертая степень
    double r = sqrt(pow(z3.first, 2) + pow(z3.second, 2));
    double theta = atan2(z3.second, z3.first);
    double r4 = pow(r, 4);
    double theta4 = 4 * theta;
    std::pair<double, double> z3_fourth_power = {r4 * cos(theta4), r4 * sin(theta4)};

    // Корень третей степени (для трех корней)
    double r3 = pow(r, 1.0/3);
    std::vector<std::pair<double, double>> z3_cube_roots;
    for (int k = 0; k < 3; k++) {
        double theta3 = (theta + 2 * M_PI * k) / 3;
        z3_cube_roots.push_back({r3 * cos(theta3), r3 * sin(theta3)});
    }

    // Выводим результаты
    std::cout << "Сложение: " << sum_z.first << " + " << sum_z.second << "i" << std::endl;
    std::cout << "Вычетание: " << diff_z.first << " + " << diff_z.second << "i" << std::endl;
    std::cout << "Умножение: " << prod_z.first << " + " << prod_z.second << "i" << std::endl;
    std::cout << "Деление: " << quot_z.first << " + " << quot_z.second << "i" << std::endl;
    std::cout << "Четвертая степень -1+i: " << z3_fourth_power.first << " + " << z3_fourth_power.second << "i" << std::endl;
    std::cout << "Корень третей степени -1+i:" << std::endl;
    for (const auto& root : z3_cube_roots) {
        std::cout << root.first << " + " << root.second << "i" << std::endl;
    }

    return 0;
}
